File: eco-core/core-plugin/src/main/java/com/willfp/ecoweapons/display/WeaponsDisplay.java
package com.willfp.ecoweapons.display;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.display.DisplayModule;
import com.willfp.eco.core.display.DisplayPriority;
import com.willfp.ecoweapons.weapons.Weapon;
import com.willfp.ecoweapons.weapons.util.WeaponUtils;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class WeaponsDisplay extends DisplayModule {
    /**
     * Create weapons display.
     *
     * @param plugin Instance of EcoWeapons.
     */
    public WeaponsDisplay(@NotNull final EcoPlugin plugin) {
        super(plugin, DisplayPriority.LOWEST);
    }

    @Override
    public void display(@NotNull final ItemStack itemStack,
                           @NotNull final Object... args) {
        ItemMeta meta = itemStack.getItemMeta();
        if (meta == null) {
            return;
        }

        Weapon weapon = WeaponUtils.getWeaponFromItem(meta);

        if (weapon == null) {
            return;
        }

        ItemMeta weaponMeta = weapon.getItem().getItemMeta();
        assert weaponMeta != null;

        List<String> lore = new ArrayList<>(weaponMeta.getLore());

        if (meta.hasLore()) {
            lore.addAll(meta.getLore());
        }
        meta.setLore(lore);
        meta.setDisplayName(weaponMeta.getDisplayName());
        meta.addItemFlags(weaponMeta.getItemFlags().toArray(new ItemFlag[0]));

        itemStack.setItemMeta(meta);
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecoweapons/display/WeaponsDisplay.kt
package com.willfp.ecoweapons.display

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.display.DisplayModule
import com.willfp.eco.core.display.DisplayPriority
import com.willfp.ecoweapons.fuels.FuelUtils
import com.willfp.ecoweapons.weapons.WeaponUtils
import org.bukkit.inventory.ItemStack

class WeaponsDisplay(plugin: EcoPlugin) : DisplayModule(plugin, DisplayPriority.LOWEST) {
    override fun display(
        itemStack: ItemStack,
        vararg args: Any
    ) {
        val meta = itemStack.itemMeta ?: return
        val weapon = WeaponUtils.getWeaponFromItem(meta)
        val fuel = FuelUtils.getFuelFromItem(meta)

        if (fuel != null) {
            val fuelMeta = fuel.itemStack.itemMeta ?: return
            val lore: MutableList<String> = fuelMeta.lore?.toMutableList() ?: return

            if (meta.hasLore()) {
                lore.addAll(meta.lore ?: return)
            }

            meta.lore = lore
            meta.setDisplayName(fuelMeta.displayName)
            itemStack.itemMeta = meta
        }

        if (weapon != null) {
            val weaponMeta = weapon.itemStack.itemMeta ?: return
            val lore: MutableList<String> = weaponMeta.lore?.toMutableList() ?: return

            if (meta.hasLore()) {
                lore.addAll(meta.lore ?: return)
            }

            meta.lore = lore
            meta.setDisplayName(weaponMeta.displayName)
            meta.addItemFlags(*weaponMeta.itemFlags.toTypedArray())
            itemStack.itemMeta = meta
        }
    }
}